code/modules/clothing/shoes/magboots.dm

```diff
@@ -58,3 +58,61 @@
 	name = "blood-red magboots"
 	icon_state = "syndiemag0"
 	magboot_state = "syndiemag"
+
+/obj/item/clothing/shoes/magboots/gravboots
+	desc = "A pair of modified magnetic boots that use an anomaly core to generate gravity around the user instead of pulling their feet to the ground. Requires a gravitational anomaly core to function."
+	name = "zero-point magboots"
+	icon_state = "gravboots0"
+	magboot_state = "gravboots"
+	slowdown_active = 1
+	var/core = FALSE
+	var/list/grav_turfs = list()
+
+/obj/item/clothing/shoes/magboots/gravboots/attackby(obj/item/C, mob/user)
+	if(istype(C, /obj/item/assembly/signaler/anomaly/grav))
+		to_chat(user, "<span class='notice'>You insert [C] into [src] and their circuitry gently hums to life.</span>")
+		core = TRUE
+		playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
+		qdel(C)
+		return
+	return ..()
+
+/obj/item/clothing/shoes/magboots/gravboots/attack_self(mob/user)
+	if(!core)
+		playsound(src.loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "<span class='warning'>[src] fail to activate without an inserted anomaly core!</span>")
+		return
+
+	if(magpulse)
+		clothing_flags &= ~NOSLIP
+		slowdown = SHOES_SLOWDOWN
+		UnregisterSignal(user, COMSIG_MOVABLE_MOVED)
+	else
+		clothing_flags |= NOSLIP
+		slowdown = slowdown_active
+		RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/distance_check)
```

Comment:
You need to unregister this on Destroy.

Reply:
Or even on `dropped`, I don't see this turned off anywhere other than the user.

Reply:
im pretty sure u dont

Reply:
but yes it does need to go away on dropped / equipped to wrong slot

Reply:
You would be able to turn these on, take them off, then keep the effect.

Reply:
> im pretty sure u dont

Might be mistaking it because it registers on user, might not need to for Destroy.